x-pack/plugins/lens/public/app_plugin/lens_top_nav.tsx

```diff
@@ -623,14 +623,16 @@ export const LensTopNavMenu = ({
             share.toggleShareContextMenu({
               anchorElement,
               allowEmbed: false,
-              allowShortUrl: false, // we'll manage this implicitly via the new service
+              allowShortUrl: true, // default is short url
```

Comment:
If only short URL is allowed for saved charts, then I need more code can be removed from Lens.
In particular the `shareableUrl` is no longer required - and all the code to generate it - and only `savedObjectURL` should be used here.

Reply:
Testing this locally I've noticed that the shared link no longer works, looping in a redirect forever:

![forever_redirecting](https://github.com/elastic/kibana/assets/924948/3e3ea884-cba0-4de1-a12a-7bdb0843efb0)

I suspect that is due to the shorting of an already shortened URL.

Reply:
> Testing this locally I've noticed that the shared link no longer works, looping in a redirect forever:
> 
> I suspect that is due to the shorting of an already shortened URL.

Thanks for pointing this out, we'd look into this!

Reply:
Fixed in [ed699bf](https://github.com/elastic/kibana/pull/180406/commits/ed699bf1600df11b0347223b75af46b1ca0e792d)